infra/examples-dev/aws-all/variables.tf

```diff
@@ -0,0 +1,251 @@
+variable "environment_name" {
+  type        = string
+  description = "friendly qualifier to distinguish resources created by this terraform configuration other Terraform deployments, (eg, 'prod', 'dev', etc)"
+
+  validation {
+    condition     = can(regex("^[a-zA-Z][a-zA-Z0-9-_ ]*[a-zA-Z0-9]$", var.environment_name))
+    error_message = "The `environment_name` must start with a letter, can contain alphanumeric characters, hyphens, underscores, and spaces, and must end with a letter or number."
+  }
+
+  validation {
+    condition     = !can(regex("^(?i)(aws|ssm)", var.environment_name))
+    error_message = "The `environment_name` cannot start with 'aws' or 'ssm', as this will name your AWS resources with prefixes that displease the AMZN overlords."
+  }
+}
+
+variable "aws_account_id" {
+  type        = string
+  description = "id of aws account in which to provision your AWS infra"
+
+  validation {
+    condition     = can(regex("^\\d{12}$", var.aws_account_id))
+    error_message = "The aws_account_id value should be 12-digit numeric string."
+  }
+}
+
+variable "aws_assume_role_arn" {
+  type        = string
+  description = "ARN of role Terraform should assume when provisioning your infra. (can be `null` if your CLI is auth'd as the right user/role)"
+  default     = null
+}
+
+variable "aws_region" {
+  type        = string
+  description = "default region in which to provision your AWS infra"
+  default     = "us-east-1"
```

Comment:
remove default to ensure is set always by customer?

Reply:
fair idea, although our bugs are normally when they set this to something *other* than `us-east-1`, but other tooling defaults to `us-east-1` (such as the npm packages used in the testing tools)